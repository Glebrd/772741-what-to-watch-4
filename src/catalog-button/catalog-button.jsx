import React from "react";
import {ActionCreator} from "../reducer";
import {connect} from "react-redux";
import PropTypes from "prop-types";

const CatalogButton = (props)=>{
  const {increaseNumberOfMovies} = props;
  return (
    <button
      onClick={increaseNumberOfMovies}
      className="catalog__button" type="button">Show more
    </button>
  );
};

const mapStateToProps = (state) => ({
  currentGenre: state.currentGenre,
});

const mapDispatchToProps = (dispatch) => ({
  increaseNumberOfMovies() {
    dispatch(ActionCreator.increaseNumberOfMoviesOnMain(8));
  },
});

CatalogButton.propTypes = {
  increaseNumberOfMovies: PropTypes.func,
};

export {CatalogButton};
export default connect(mapStateToProps, mapDispatchToProps)(CatalogButton);
