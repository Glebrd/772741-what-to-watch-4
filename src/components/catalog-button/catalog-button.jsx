import React from "react";
import {ActionCreator} from "../../reducer/application/application";
import {connect} from "react-redux";
import PropTypes from "prop-types";
import {getCurrentGenre} from "../../reducer/application/selectors";

const CatalogButton = (props)=>{
  const {increaseNumberOfMovies} = props;
  return (
    <button
      onClick={increaseNumberOfMovies}
      className="catalog__button" type="button">Show more
    </button>
  );
};

const mapStateToProps = (state) => ({
  currentGenre: getCurrentGenre(state),
});

const mapDispatchToProps = (dispatch) => ({
  increaseNumberOfMovies() {
    dispatch(ActionCreator.increaseNumberOfMoviesOnMain(8));
  },
});

CatalogButton.propTypes = {
  increaseNumberOfMovies: PropTypes.func,
};

export {CatalogButton};
export default connect(mapStateToProps, mapDispatchToProps)(CatalogButton);
