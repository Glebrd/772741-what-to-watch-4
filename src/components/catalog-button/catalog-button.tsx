import * as React from "react";
import {ActionCreator} from "../../reducer/application/application";
import {connect} from "react-redux";
import {getCurrentGenre} from "../../reducer/application/selectors";

interface Props {
  onCatalogButtonClick: () => {};
}

const CatalogButton: React.FunctionComponent<Props> = (props: Props) => {
  const {onCatalogButtonClick} = props;
  return (
    <button
      onClick={onCatalogButtonClick}
      className="catalog__button" type="button">Show more
    </button>
  );
};

const mapStateToProps = (state) => ({
  currentGenre: getCurrentGenre(state),
});

const mapDispatchToProps = (dispatch) => ({
  onCatalogButtonClick() {
    dispatch(ActionCreator.increaseNumberOfMoviesOnMain(8));
  },
});

export {CatalogButton};
export default connect(mapStateToProps, mapDispatchToProps)(CatalogButton);
