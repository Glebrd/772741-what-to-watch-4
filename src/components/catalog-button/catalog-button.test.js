import React from "react";
import renderer from "react-test-renderer";
import {CatalogButton} from "./catalog-button";

it(`Catalog genres list renders`, () => {
  const tree = renderer
    .create(
        <CatalogButton/>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
