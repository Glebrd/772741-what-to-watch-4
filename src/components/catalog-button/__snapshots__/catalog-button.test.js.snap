// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalog genres list renders 1`] = `
<button
  className="catalog__button"
  type="button"
>
  Show more
</button>
`;
